### PR TITLE
Cleanup and CentOS 7.2 fixes

### PR DESCRIPTION
**Pull request by RyanJarv**
Going to hold off merging this atm, would like to do some more testing and also see if there is a better way to handle the hostname thing (It also comes up when the hostname changes).

- Varnish expects the varnish instance name to be `hostname` by default however this is sometimes different then `hostname -s` which is used by `ohai hostname`. This seemed to only be an issue on CentOS 7.2.
- Use /etc/varnish/varnish.params for systemd init file. This is what the package uses and I think will remove some confusion about how settings are set. https://github.com/sous-chefs/varnish/issues/103
- Remove `/usr/share/varnish/reload-vcl` on debian. This was added to support the '-j' option in varnish 4.1 however it has been fixed upstream.
- Few RuboCop fixes.

### CODE REVIEW

**Comment by RyanJarv** [comment]
The main issue with the hostname is that the varnishadm command can't find the _.vsm file. I would imagine there's a way to configure what varnishadm looks for by default but so far haven't found it or haven't dug into it enough.

**Comment by RyanJarv** [comment]
So the last thing with the hostname changing doesn't seem to be an issue for me anymore. Will go ahead fix that later if it comes up again. Updated the minor version since we're removing the init template here.

Unless other people start running into this issue I would like to get https://github.com/sous-chefs/varnish/pull/121 finished before merging but other then that I think it should be good.

**Comment by RyanJarv** [comment]
Ignore me, thought the build was broken till we got that merged looks like the README just needs to be updated. This should be good to merge if everyone is ok with it (not sure how many reviewers we want on merges).

File: templates/default/default.erb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #
 # CHEF MANAGED FILE: DO NOT EDIT
-# Generated by Chef for <%= node['hostname'] %>
+# Generated by Chef for <%= `hostname`.strip %>
```
Comment (tas50) [inline templates/default/default.erb]: The hostname line here should just get nuked entirely

File: test/fixtures/cookbooks/install_varnish/recipes/full_stack.rb [hunk 2]
```diff
@@ -19,10 +19,9 @@
   variables(
     config: {
       backend_host: '127.0.0.1',
-      backend_port: '8080',
+      backend_port: '8080'
```
Comment (tas50) [inline test/fixtures/cookbooks/install_varnish/recipes/full_stack.rb]: You'll need that comma to pass cookstyle

Comment (RyanJarv) [inline test/fixtures/cookbooks/install_varnish/recipes/full_stack.rb]: Master is still running CircleCI and rubocop currently, had to remove that to get it to pass (https://circleci.com/gh/sous-chefs/varnish/709). We'll want to get that in line with the other cookbooks but will probably have to do that in another pull request, updated the link on the README though so at least that isn't broken anymore.

File: templates/default/default.erb (hunk 1 above)
Comment (RyanJarv) [inline templates/default/default.erb]: Yeah not really sure what that's good for, removed it.

**Comment by rshade** [comment]
👍
